modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalAddFilter.tsx

```diff
@@ -297,10 +339,15 @@ export function ModalAddFilter({
 						emptyStateMessage={Liferay.Language.get(
 							'there-are-no-columns-available'
 						)}
+						error={errors.selectedFilterBy}
 						items={filteredAvailableFields}
 						label={Liferay.Language.get('filter-by')}
 						onChangeQuery={setQuery}
-						onSelectItem={setSelectedFilterBy}
+						onSelectItem={(item) => {
+							setSelectedFilterBy(item);
+							setSelectedFilterType(undefined);
```

Comment:
Hi @danvb112. Once Milton reviewed my code and said that there's a difference between value null and undefined. I think in this case you should set to null, according to this definition (https://www.tektutorialshub.com/typescript/null-undefined-in-typescript/#:~:text=TypeOf-,Undefined%20%26%20Null,does%20not%20have%20a%20value.).
But please, feel free to make the change or not. You can ask Milton in the future hehe

> The value undefined means value is not assigned & you don’t know its value. It is an unintentional absence of value. It means that a variable has been declared but has not yet been assigned a value.

> The value null indicates that you know that the field does not have a value. It is an intentional absence of value.

Reply:
Thanks @alinedoleron